### PR TITLE
#19 Select Seats + Postgres

### LINKED ISSUE #19: Select Seats + Postgres
## Expected Behaviour

When reserving seats, we need to POST them to the backend and add the reserved seats to our DB.

## Checklist

_Detail the steps below to help us reproduce the problem you encountered._

- [x] Hook up route to database
- [x] get the current booked seats
- [x] merge the booked seats with the new booked seats

## Hints
- Booked seats are stored as a comma seperated string (e.g. "A1,A2")
- You need to transform that into and array to update it

### PR DESCRIPTION
**Pull request by focusedXmind**
## Checklist

Confirm you have completed the following actions prior to submitting this PR.

- [X] There is an existing issue report for this PR.
- [X] I have created a feature branch.
- [X] My changes have been committed.
- [X] I have pushed my changes to the branch.

## Description

When reserving seats, we need to POST them to the backend and add the reserved seats to our DB.

The task is about the modification of the functionality for reserving seats with postgres. The following points to do within the task were:

- hook up route to database
- get the current booked seats
- merge the booked seats with the new booked seats

Closes #19

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,6 +24,7 @@
     "@prisma/client": "^5.17.0",
     "dotenv": "^16.4.5",
     "express": "^4.19.2",
+    "npm": "^10.8.2",
```
Comment (dnmct) [inline package.json]: why did you install npm locally?

File: prisma/seed.ts [hunk 2]
```diff
@@ -1,69 +1,26 @@
-import { connect } from "http2";
-import { prisma } from "../lib/db";
-import { User, Prisma } from "@prisma/client";
+import { PrismaClient } from "@prisma/client";
 
-async function main() {
-  await prisma.user.deleteMany();
-  await prisma.bookmark.deleteMany();
+const prisma = new PrismaClient();
```
Comment (louziferr) [inline prisma/seed.ts]: please import prisma from "../lib/db" and don't create a new PrismaClient

File: prisma/seed.ts
Comment (louziferr) [inline prisma/seed.ts:1]: Please also seed users and bookmarks (just like in the main branch).

File: prisma/seed.ts [hunk 3]
```diff
@@ -1,69 +1,26 @@
...
-    {
-      movieId: "693134",
-      user: { connect: { id: createdUsers[1].id } },
-    },
-    {
-      movieId: "693134",
-      user: { connect: { id: createdUsers[0].id } },
-    },
-  ];
+  await Promise.all(
```
Comment (louziferr) [inline prisma/seed.ts]: Please loop over seats with a for-loop and then await prisma.seat.create(...) . Just like with users in the main branch.


Please also seed ScreeningSeats.

File: routes/reservations.ts [hunk 4]
```diff
@@ -1,63 +1,57 @@
...
-  const dbFile = fs.readFileSync("./db.json", { encoding: "utf-8" });
-  return JSON.parse(dbFile) as DB;
-}
-
-function saveDB(db: DB) {
-  fs.writeFileSync("./db.json", JSON.stringify(db, null, 2));
-}
+import { PrismaClient } from "@prisma/client";
 
+const prisma = new PrismaClient();
```
Comment (louziferr) [inline routes/reservations.ts]: Please import prisma from "../lib/db". Don't create a new PrismaClient.

File: routes/reservations.ts [hunk 5]
```diff
@@ -1,63 +1,57 @@
...
+    });
 
-  // Check for already booked seats
-  const alreadyBookedSeats = seats.filter((seat: string) =>
-    screening.bookedSeats.includes(seat)
-  );
+    if (!screening) {
+      return res.status(404).json({ message: "Screening not found" });
+    }
+
```
Comment (louziferr) [inline routes/reservations.ts:23]: Please also check if the seat exists.

**Review by louziferr — CHANGES_REQUESTED** [review]
changes needed

**Review by dnmct — APPROVED** [review]

**Pull request merged**